### PR TITLE
Add methods and script to cache all Faculty CV data

### PR DESCRIPTION
**Pull request by tlemburg**
Queries to Digital Measures were causing epic server load,
especially on digital signage where offices/departments
were being printed. Instead of getting this data on demand,
this commit adds a script which will be added to cron to
grab the data every 15 minutes from Digital Measures, and
cache it for fast retrieval on all queries.

This method can be tested by following the directions given
in examples/cache_example.txt .

### CODE REVIEW

**Comment by mfairchild365** [comment]
My initial thoughts:

1. Should `getCategory()` be modified? I think it should only retrieve the cache and return nothing if there is not a cached entry rather than performing a new API call
2. How are cleaning old cached data? If a user is removed from DM, how is the cached entry removed from directory?

**Comment by tlemburg** [comment]
1. Probably not. If we change to that method, a new record entered into DM will not show up at all for 15 minutes in directory, which isn't terrible, but kind of kills the functionality of the original code. It's ok to still request the info on a case-by-case basis, should a slab of memcache get erased for whatever reason. This code will almost never get run but it is good to have there as a backup in case.

2. Cleaning of old cached data isn't being done right now, but the life of all data in the cache is only 20 minutes. I realize this sort of contradicts my point above, but there is a difference as making change 1 would change the user experience, whereas leaving change 2 the way it is doesn't affect the user at all.

TL;DR: 1) no. 2) we aren't. 20 minutes TTL.
